lib/post_data.js

```diff
@@ -52,20 +52,29 @@ SlackDataPostHandler.prototype.postData = function(data) {
   split_message = utils.splitMessage(this.formatter.formatData(data.message));
 
   if (split_message.text) {
-    attachment = {
+    var content = {
       color: attachment_color,
-      text: split_message.text,
       "mrkdwn_in": ["text", "pretext"],
-      fallback: split_message.text
     };
     if (data.extra && data.extra.slack) {
-      for (var attrname in data.extra.slack) { attachment[attrname] = data.extra.slack[attrname]; }
+      for (var attrname in data.extra.slack) { content[attrname] = data.extra.slack[attrname]; }
     }
-    this.robot.emit('slack-attachment', {
-      channel: recipient,
-      text: pretext + split_message.pretext,
-      content: attachment
-    });
+    var robot = this.robot;
+    var chunks = split_message.text.match(/[\s\S]{1,7900}/g);
+    var sendChunk = function (i) {
```

Comment:
I think it would be good to make this a top-level function so it's easier / possible to test it.

To make testing easier I would probably also get rid of some global args and pass them to (and return from) the new function instead.